### PR TITLE
Provide resource hook to mutate scope after filters are applied.

### PR DESCRIPTION
**Pull request by zeisler**
In order to do scope manipulation only after all provided filters are applied adding a hook `#after_filtering(scope)`. The resulting scope will be used for sorting, pagination, and stats. This might be useful for doing a group by or some other one time query where the filtered scope is used as the sub query.

### CODE REVIEW

**Comment by richmolj** [comment]
I'm not totally against it, but how feasible is it to use the existing `#around_scoping` for this?

**Comment by zeisler** [comment]
The order of operation in `#around_scoping` is `DefaultFilter` -> `Filter` -> `Sort` -> `Paginate`. Where the scope is mutated and pass along at each phase. Then later the scope is passed to `stats`. I need something to effect everything downstream of `Filter` I don't see a way to do that with `#around_scoping`.

**Comment by richmolj** [comment]
Apologies for the delay @zeisler - I think this looks great. My only thing is the `after_filters` name...what about `after_filtering`? It just currently strikes me as too close to "add an 'after filter'" versus "this is a hook that runs after filtering".

File: lib/graphiti/resource.rb [hunk 1]
```diff
@@ -20,6 +20,10 @@ def around_scoping(scope, query_hash)
       yield scope
     end
 
+    def after_filters(scope)
```
Comment (zeisler) [inline lib/graphiti/resource.rb]:
```suggestion
    def after_filtering(scope)
```

File: lib/graphiti/scoping/filter.rb [hunk 2]
```diff
@@ -16,7 +16,7 @@ def apply
         @scope = filter_scope(filter, operator, value)
       end
 
-      @scope
+      resource.after_filters(@scope)
```
Comment (zeisler) [inline lib/graphiti/scoping/filter.rb]:
```suggestion
      resource.after_filtering(@scope)
```

File: spec/filtering_spec.rb [hunk 3]
```diff
@@ -1677,4 +1677,23 @@ def admin?
       end
     end
   end
+
+  context "when calling #after_filters hook" do
+    before do
+      resource.class_eval do
+        def after_filters(scope)
```
Comment (zeisler) [inline spec/filtering_spec.rb]:
```suggestion
        def after_filtering(scope)
```

File: spec/filtering_spec.rb [hunk 4]
```diff
@@ -1677,4 +1677,23 @@ def admin?
       end
     end
   end
+
+  context "when calling #after_filters hook" do
```
Comment (zeisler) [inline spec/filtering_spec.rb]:
```suggestion
  context "when calling #after_filtering hook" do
```

**Comment by zeisler** [comment]
@richmolj I've made the changes you've recommended.

**Pull request merged**